pkg/pipeline/transform/transform_network.go

```diff
@@ -119,7 +121,33 @@ func (n *Network) Transform(inputEntry config.GenericMap) (config.GenericMap, bo
 			}
 			kubernetes.EnrichLayer(outputEntry, rule.KubernetesInfra)
 		case api.OpReinterpretDirection:
-			reinterpretDirection(outputEntry, &n.DirectionInfo)
+			isInner, isSrcReporter := reinterpretDirection(outputEntry, &n.DirectionInfo)
+			if !isInner && n.cache != nil {
+				if isSrcReporter {
+					// add flow to cache and stop pipeline for now
+					err := n.cache.PutDedupInnerToCache(outputEntry)
+					if err != nil {
+						log.Errorf("error putting cache %v for %v", err, outputEntry)
+					}
+					return nil, false
+				}
+
+				found, err := n.cache.GetDedupInner(outputEntry)
+				if err != nil {
+					log.Errorf("error getting cache %v for %v", err, outputEntry)
+				} else if found != nil {
+					// append interfaces and direction to flow
+					srcFlow := *found
+					log.Debugf("merging cached flow %v with %v ...", srcFlow, outputEntry)
+					if len(srcFlow["Interfaces"].([]interface{})) == len(srcFlow["IfDirections"].([]interface{})) {
+						for i := 0; i < len(srcFlow["Interfaces"].([]interface{})); i++ {
+							outputEntry["Interfaces"] = append(outputEntry["Interfaces"].([]string), fmt.Sprintf("%s*", srcFlow["Interfaces"].([]interface{})[i].(string)))
+							outputEntry["IfDirections"] = append(outputEntry["IfDirections"].([]int), int(srcFlow["IfDirections"].([]interface{})[i].(float64)))
+						}
+					}
+					log.Debugf("merged flow: %v", outputEntry)
```

Comment:
:point_up_2: merging `Egress` flow content in `Ingress` one.
For now just merging `Interfaces` and `IfDirections` and adding a `*` next to the names.
We could imagine merging more feature related contents, TCP flags and also check Bytes / Packets amounts

Reply:
so if Egress comes after Ingress, it will be ignored? Perhaps it's another reason to not have the timestamp in the key as discussed [here](https://github.com/netobserv/flowlogs-pipeline/pull/639/files#r1530003170)

Reply:
yeah currently if `Egress` comes after `Ingress`, it's pushed in the cache and never retreived.
Only `Ingress` will appear in storage and will not have merged fields.

An improvment could be to send first flow to the cache whatever direction is found and merge on the "second" one; but that will double the cache queries